Snowflake.Data/Core/Session/EasyLoggingStarter.cs

```diff
@@ -90,17 +113,53 @@ private string GetLogPath(string logPath)
             var logPathOrDefault = logPath;
             if (string.IsNullOrEmpty(logPath))
             {
-                s_logger.Warn("LogPath in client config not found. Using temporary directory as a default value");
-                logPathOrDefault = Path.GetTempPath();
+                s_logger.Warn("LogPath in client config not found. Using home directory as a default value");
+                try
+                {
+                    logPathOrDefault = _environmentOperations.GetFolderPath(Environment.SpecialFolder.UserProfile);
+                    if (string.IsNullOrEmpty(logPathOrDefault))
+                    {
+                        throw new Exception("No log path found for easy logging. Home directory is not configured and log path is not provided");
+                    }
+                }
+                catch(Exception e)
+                {
+                    var errorMessage = $"Error while trying to retrieve the home directory: {e}";
+                    s_logger.Error(errorMessage);
+                    throw new Exception(errorMessage, e);
+                }
             }
             var pathWithDotnetSubdirectory = Path.Combine(logPathOrDefault, "dotnet");
             if (!_directoryOperations.Exists(pathWithDotnetSubdirectory))
             {
-                _directoryOperations.CreateDirectory(pathWithDotnetSubdirectory);
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    _directoryOperations.CreateDirectory(pathWithDotnetSubdirectory);
+                }
+                else
+                {
+                    _unixOperations.CreateDirectoryWithPermissions(pathWithDotnetSubdirectory,
+                        FilePermissions.S_IRUSR | FilePermissions.S_IWUSR | FilePermissions.S_IXUSR);
+                    CheckDirPermissionsOnlyAllowUser(pathWithDotnetSubdirectory);
+                }
             }
 
             return pathWithDotnetSubdirectory;
         }
+
+        private void CheckDirPermissionsOnlyAllowUser(string dirPath)
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                return;
+
+            _unixOperations.SetDirInfo(dirPath);
+            var dirPermissions = _unixOperations.GetDirPermissions();
+            if (dirPermissions != FileAccessPermissions.UserReadWriteExecute)
```

Comment:
if the config file would be located in a direcotry to which the user cannot write it should also be OK

Reply:
Ok, will modify that check

Reply:
Fixed